src/tests/zfs/zfs_custom_node_topology/zfs_custom_node_topology_test.go

```diff
@@ -0,0 +1,397 @@
+	Then All volumes should be in pending state
+	When Set the ALLOWED_TOPOLOGIES env in zfs node-daemonset with custom topology key
+	Then Verify topology key is now available in csi node for local.csi.openebs.io plugin driver
+	And All volumes should be transition from pending to bound state
+	And All volumes should get provisioned on only those node which was labeled prior to the provisioning
+*/
+
+func customTopologyWfcTest(decor string, engine common.OpenEbsEngine, volType common.VolumeType, fstype common.FileSystemType, volBindModeWait bool) {
+	appInstances = []*k8stest.FioApplication{}
+	workerNodes, err := zfs.ListZfsNode(common.NSOpenEBS())
+	Expect(err).ToNot(HaveOccurred(), "failed to list worker node")
+
+	// minimum worker nodes in cluster should be two
+	Expect(len(workerNodes)).Should(BeNumerically(">=", 2),
+		"test case requires are least 2 worker nodes, %d nodes found", len(workerNodes))
+
+	key = "zfse2e/nodename"
+	targetNode = workerNodes[0]
+
+	// label worker node
+	err = k8stest.LabelNode(targetNode, key, targetNode)
+	Expect(err).ToNot(HaveOccurred(), "failed to label node %s", targetNode)
+
+	zfsScOptions := k8stest.ZfsOptions{
+		//PoolName:      nodeConfig.PoolName,
+		PoolName:      "zfspv-pool",
+		ThinProvision: common.No,
+		RecordSize:    "128k",
+		Compression:   common.Off,
+		DedUp:         common.Off,
+		AllowedTopologies: []coreV1.TopologySelectorTerm{
+			{
+				MatchLabelExpressions: []coreV1.TopologySelectorLabelRequirement{
+					{
+						Key:    key,
+						Values: []string{targetNode},
+					},
+				},
+			},
+		},
+	}
+
+	for i := 0; i <= len(workerNodes); i++ {
+		app := k8stest.FioApplication{
+			Decor:                   fmt.Sprintf("%s-%d", decor, i),
+			VolSizeMb:               1024,
+			OpenEbsEngine:           engine,
+			VolType:                 volType,
+			FsType:                  fstype,
+			Loops:                   1,
+			VolWaitForFirstConsumer: volBindModeWait,
+			Zfs:                     zfsScOptions,
+		}
+		appInstances = append(appInstances, &app)
+	}
+
+	// deploy all zfs volume
+	for ix, app := range appInstances {
+		logf.Log.Info(fmt.Sprintf("%d)", ix), "zfs-volume", app.Decor)
+		err = app.CreateVolume()
+		Expect(err).ToNot(HaveOccurred(), "failed to create volume %s, %v", app.Decor, err)
+	}
+
+	for ix, app := range appInstances {
+		logf.Log.Info(fmt.Sprintf("%d)", ix), "zfs-volume", app.Decor)
+		// verify pvc to be in pending state
+		pvcPhase, err := k8stest.GetPvcStatusPhase(app.GetPvcName(), common.NSDefault)
+		Expect(err).ToNot(HaveOccurred(), "failed to get pvc phase")
+		Expect(pvcPhase).Should(Equal(coreV1.ClaimPending), "pvc phase is not pending")
+	}
+
+	logf.Log.Info("Sleep for 10 MIN")
+	time.Sleep(10 * time.Minute)
+
+	productConfig := e2e_config.GetConfig().Product
+	label := fmt.Sprintf("%s=%s", productConfig.LocalEngineComponentPodLabelKey,
+		productConfig.ZfsEngineComponentDsPodLabelValue)
+
+	// restart zfs daemonset pods
```

Comment:
Why the restart of daemon set pods?

Reply:
After applying node label with key, we need to restart to daemonset so that csinode kubernetes object for local.csi.openebs.io plugin driver picks the key